spec/javascripts/mobile/mobile_post_actions_spec.js

```diff
@@ -0,0 +1,226 @@
+describe("Diaspora.Mobile.PostActions", function(){
+  describe("initialize", function(){
+    beforeEach(function(){
+      spec.loadFixture("aspects_index_mobile_public_post");
+      spyOn(Diaspora.Mobile.PostActions, "onLike");
+      spyOn(Diaspora.Mobile.PostActions, "onReshare");
+      Diaspora.Mobile.PostActions.initialize();
+    });
+
+    it("binds the events", function(){
+      $(".stream .like-action").trigger("tap");
+      expect(Diaspora.Mobile.PostActions.onLike).toHaveBeenCalled();
+      $(".stream .like-action").click();
+      expect(Diaspora.Mobile.PostActions.onLike).toHaveBeenCalled();
+      $(".stream .reshare-action").trigger("tap");
+      expect(Diaspora.Mobile.PostActions.onReshare).toHaveBeenCalled();
+      $(".stream .reshare-action").click();
+      expect(Diaspora.Mobile.PostActions.onReshare).toHaveBeenCalled();
+    });
+  });
+
+  describe("toggleActive", function(){
+    beforeEach(function(){
+      spec.loadFixture("aspects_index_mobile_public_post");
+      Diaspora.Mobile.PostActions.initialize();
+      this.link = $(".stream .like-action").first();
+    });
+
+    it("toggles active and inactive classes", function(){
+      expect(this.link).toHaveClass("inactive");
+      expect(this.link).not.toHaveClass("active");
+      Diaspora.Mobile.PostActions.toggleActive(this.link);
+      expect(this.link).not.toHaveClass("inactive");
+      expect(this.link).toHaveClass("active");
+      Diaspora.Mobile.PostActions.toggleActive(this.link);
+      expect(this.link).toHaveClass("inactive");
+      expect(this.link).not.toHaveClass("active");
+    });
+  });
+
+  describe("onLike", function(){
+    beforeEach(function(){
+      spec.loadFixture("aspects_index_mobile_public_post");
+      spyOn(Diaspora.Mobile.PostActions, "like");
+      spyOn(Diaspora.Mobile.PostActions, "unlike");
+      Diaspora.Mobile.PostActions.initialize();
+      this.link = $(".stream .like-action").first();
+    });
+
+    it("adds and removes loading class", function(){
+      expect(this.link).not.toHaveClass("loading");
+      Diaspora.Mobile.PostActions.showLoader(this.link);
+      expect(this.link).toHaveClass("loading");
+      Diaspora.Mobile.PostActions.hideLoader(this.link);
+      expect(this.link).not.toHaveClass("loading");
+    });
+
+    it("doesn't do anything if link is loading", function(){
+      Diaspora.Mobile.PostActions.showLoader(this.link);
+      expect(this.link).toHaveClass("loading");
```

Comment:
Since we already test above that `showLoader` does the right thing, you could also add the `loading` class to the link manually. Then this test would also work if the `showLoader` function breaks.